bundles/AdminBundle/Resources/public/js/pimcore/document/document.js

```diff
@@ -175,63 +176,69 @@ pimcore.document.document = Class.create(pimcore.element.abstract, {
         pimcore.elementservice.deleteElement(options);
     },
 
+    close: function() {
+        var tabPanel = Ext.getCmp("pimcore_panel_tabs");
+        tabPanel.remove(this.tab);
+    },
+
     saveClose: function (only) {
         this.save(null, only, function () {
-            var tabPanel = Ext.getCmp("pimcore_panel_tabs");
-            tabPanel.remove(this.tab);
+            this.close();
         }.bind(this));
     },
 
     publishClose: function () {
         this.publish(null, function () {
-            var tabPanel = Ext.getCmp("pimcore_panel_tabs");
-            tabPanel.remove(this.tab);
+            this.close();
         }.bind(this));
     },
 
     publish: function (only, callback) {
-        this.data.published = true;
+        this.save("publish", only, callback, function (rdata) {
```

Comment:
update page_snippet::publish to pass successCallback to $super https://github.com/pimcore/pimcore/blob/b82377b72d2b1d9deb4f1db8c9e9a74b72ff911e/bundles/AdminBundle/Resources/public/js/pimcore/document/page_snippet.js#L450

Reply:
Is this really needed? The signature of `publish` hasn't changed.

Reply:
you are right..only changing save should be sufficient.